core/unit_test/TestCTestDevice.cpp

```diff
@@ -0,0 +1,123 @@
+#include <gtest/gtest.h>
+
+namespace Kokkos {
+namespace Impl {
+
+int get_ctest_gpu(const char* local_rank_str);
+
+}  // namespace Impl
+}  // namespace Kokkos
+
+// Needed because https://github.com/google/googletest/issues/952 has not been
+// resolved
+#define EXPECT_THROW_WITH_MESSAGE(stmt, etype, whatstring) \
+  EXPECT_THROW(try { stmt; } catch (const etype& ex) {     \
+    EXPECT_EQ(whatstring, std::string(ex.what()));         \
+    throw;                                                 \
+  },                                                       \
+               etype)
```

Comment:
```diff
diff --git a/core/unit_test/TestCTestDevice.cpp b/core/unit_test/TestCTestDevice.cpp
index a925c9f..350ee3b 100644
--- a/core/unit_test/TestCTestDevice.cpp
+++ b/core/unit_test/TestCTestDevice.cpp
@@ -11,11 +11,12 @@ int get_ctest_gpu(const char* local_rank_str);
 // Needed because https://github.com/google/googletest/issues/952 has not been
 // resolved
 #define EXPECT_THROW_WITH_MESSAGE(stmt, etype, whatstring) \
-  EXPECT_THROW(try { stmt; } catch (const etype& ex) {     \
-    EXPECT_EQ(whatstring, std::string(ex.what()));         \
-    throw;                                                 \
-  },                                                       \
-               etype)
+  EXPECT_THROW(                                            \
+      try { stmt; } catch (const etype& ex) {              \
+        EXPECT_EQ(whatstring, std::string(ex.what()));     \
+        throw;                                             \
+      },                                                   \
+      etype)
 
 class ctest_environment : public ::testing::Test {
  protected:
```
You need to apply `clang-format` here as well.

Reply:
I did use `clang-format` and this is what it gave me. You said 8.0 right? Did I miss a config file?

Reply:
Hmmm... so using `git-clang-format` produced the incorrect formatting currently in the commit, but `clang-format` produced the correct formatting. Looks like `git-clang-format` is not reading the config file or something.